gdnative-core/src/init/macros.rs

```diff
@@ -21,3 +21,77 @@ macro_rules! godot_init {
         };
     };
 }
+
+/// Legacy macro that declares all the API endpoints necessary to initialize a NativeScript library.
+///
+/// This is a shim for the new [`#[gdnative::init::callbacks]`][crate::init::callbacks] attribute
+/// macro, which is now used to declare API callbacks along with the
+/// [`GDNativeCallbacks`][crate::init::GDNativeCallbacks] trait.
+#[macro_export]
+#[deprecated = "use the #[gdnative::init::callbacks] attribute macro instead"]
+macro_rules! godot_nativescript_init {
+    () => {
+        const _: () = {
+            struct GDNativeCallbacksImpl;
+
+            #[$crate::init::callbacks]
+            unsafe impl $crate::init::GDNativeCallbacks for GDNativeCallbacksImpl {}
+        };
+    };
+    ($callback:ident) => {
+        const _: () = {
+            struct GDNativeCallbacksImpl;
+
+            #[$crate::init::callbacks]
+            unsafe impl $crate::init::GDNativeCallbacks for GDNativeCallbacksImpl {
+                fn nativescript_init(handle: $crate::init::InitHandle) {
+                    $callback(handle);
+                }
+            }
+        };
+    };
+    (_ as $fn_name:ident) => {
+        ::core::compile_error!("this syntax is no longer supported. use the #[gdnative::init::callbacks] attribute macro with a prefix instead");
```

Comment:
Do we explicitly use `core` and not `std`? The distinction should only matter in `no_std` environments, something we anyway can't provide 🤔

Reply:
Regarding `no_std`, I don't actually remember there being anything from `std` proper (not in `alloc`) that we depend for core functionalities. It should be irrelevant for all the openly available platforms, but perhaps it can be slightly relevant for consoles?

Of course it's also fine here to just use `std`, if you think that's better.

Reply:
It's mostly a matter of consistency and surprise factor -- when I see `core::`, my association is immediately `no_std`, but maybe the Rust community sees this differently, and using `core` when it suffices is good practice.

Most of the codebase currently uses `std::` though, with a handful of `core::` occurrences for things like `core::cmp::Ord`, `core::marker::PhantomData` -- but even those are not used consistently. It's only a cosmetic issue though.

Reply:
I understand. For this PR, I'll switch to `std`. I've created an issue for investigating `no_std` later: #1028